### ISSUE TITLE
Sidebar dismissible works only if Sidebar is modal

### ISSUE THREAD

**Issue by xAl3xFx** [primefaces/primereact#2562]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Current behavior**
When using `dismissible` prop on the sidebar it only works when the Sidebar has the `modal` property.

**Expected behavior**
Using `dismissible` should work without having modal Sidebar as well.

* **PrimeReact version:**
7.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Chrome
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by melloware** [comment]
PR Submitted.

**Comment by melloware** [comment]
Also I am going to submit another one because `dismissable` is spelled wrong everywhere its `dismissible`.

**Comment by inad9300** [comment]
I have encountered what seems to be the issue described here. In the following link, you can see how toggling dismissibility doesn't actually immediately toggle it. It is only after closing and reopening the sidebar that the dismissible flag is taken into account (so it's probably related to issue https://github.com/primefaces/primereact/issues/1065 too).

https://codesandbox.io/s/primereact-test-forked-ll7b2p?file=/src/index.js

**Comment by inad9300** [comment]
@melloware mentioning you in case my message wasn't seen, since this is a closed issue 🙃

**Comment by melloware** [comment]
@inad9300 can you open a new ticket so we can track it.  You can reference this issue and include your reproducer.

**Comment by melloware** [comment]
@inad9300 I have a fix if you want to open a new ticket.
